---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  out.width = "100%"
)
```

# AugEITD

<!-- badges: start -->
<!-- badges: end -->

This package provides users a function to determine a training set for genomic selection. The aim for this training set is to train the GBLUP model finding the best genotype from large candidate set.
Augmented EI presented by [Tsai et al.(2021)](https://doi.org/10.1007/s13253-021-00454-2) is used as the criteria and NDCG (Normalized dicounted cumulated gain, Blondel et al.(2015)) is used as the index of evaluating model prediction accuracy in this package.

For more details of methods in this package, please check the [article].

Users will obtain :
1. the selection index of genotypes
which indicates to the priority order of genotypes being selected as a training set. The ranking is based on the augmented EI value of each genotype. 
2. the evaluation result of the optimal training set
3. the relative efficiency of the optimal training set at different sizes to the whole dataset based on collected phenotype data

## Installation

The development version of AugEITD could be download from [GitHub](https://github.com/huining0312/AUGEITD) with:

``` r
# install.packages("devtools")
# library(devtools)
install_github("huining0312/AUGEITD",dependencies=TRUE,force=TRUE)
```

## Example
Two datasets are provided for users. The tropical rice dataset is provided by [Spindel et al. (2015)](https://doi.org/10.1371/journal.pgen.1004982) and the 44K rice dataset is published by [Zhao et al. (2011)](https://doi.org/10.1038/ncomms1467). 
The kinship matrix generated by arranged SNP marker matrix are provided in this package; while the raw data are available at [Data Dryad digital repository](https://datadryad.org/stash/dataset/doi:10.5061/dryad.7369p) and [Rice Diversity website](http://www.ricediversity.org/data/)  respectively.

```{r example}
library(AugEITD)
# dataset without strong population structure
#data("geno_trop")
#result = gen_sel_Index(kinship = geno_trop,nsim = 1000,n=nrow(geno_trop))

# dataset with strong population structure
#data("geno_rice44K")
#data("subpop_tag")
#result = gen_sel_Index(kinship = geno_rice44K,nsim = 1000,n=nrow(geno_rice44K),subpopTag = subpop_tag)
```

## Authors
* Hui-Ning Tu
  + Author, maintainer
  + Email: [huining0312@gmail.com](mailto:huining0312@gmail.com)
* Chen-Tuo Liao
  + Author, thesis advisor
  + Email:  [ctliao@ntu.edu.tw](mailto:ctliao@ntu.edu.tw)
  + Department of Agronomy, National Taiwan University, Taipei, Taiwan

